pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/opcua/OpcUaClientDriver.java

```
@@ -0,0 +1,211 @@
+
+import static org.eclipse.milo.opcua.stack.core.types.builtin.unsigned.Unsigned.uint;
+import static org.eclipse.milo.opcua.stack.core.util.ConversionUtil.toList;
+
+public class OpcUaClientDriver extends SimpleMemorylessDriver<OpcUaRawData> {
+
+    private static final Logger log = LoggerFactory.getLogger(OpcUaClientDriver.class);
+    private static final Gson jsonParser = new Gson();
+
+    private static String ENDPOINT = "ENDPOINT";
+    private static String NS_INDEX = "NAMESPACE_INDEX";
+    private static String NODE_ID  = "NODE_IDENTIFIER";
+    private static String NODE_FILTER  = "NODE_FILTER_REGEX";
+    private static OpcUaClient opcUaClient;
+    private static Pattern nodeFilter;
+    private static List<NodeId> sensorList ;
+    private static List<ReadValueId> readValueIds;
+    private static NamespaceTable ns;
+
+    public OpcUaClientDriver(DeviceDriverConfig config) {
+        super(config);
+        try {
+            log.info("Trying to establish connection with OPC server");
+            //TODO :connection monitoring and restore
+            opcUaClient = buildClient();
+            opcUaClient.connect().get();
+            log.info("Connection established with OPC server");
+            ns = opcUaClient.getNamespaceTable();
+            log.info("Creating sensor List");
+            nodeFilter = Pattern.compile(getNodeFilter());
+            sensorList = new LinkedList<>();
+            if (opcUaClient.getAddressSpace().getNode(getNodeID()).getNodeClass().getValue() == NodeClass.Variable.getValue())
+            {
+                sensorList.add(getNodeID());
+            }
+            filterNode(opcUaClient,getNodeID());
+            readValueIds = sensorList.stream().map(nodeId -> new ReadValueId(nodeId, AttributeId.Value.uid(), null, null)).collect(Collectors.toUnmodifiableList());
+        } catch (UaException | InterruptedException | ExecutionException e) {
+            log.error("Error in connection to UA Server", e);
```

This is catching an error and logging it but otherwise ignoring it. It would be better to not catch the error at all.

---

Constructor now throws the exception to the caller rather than suppressing it with logging,